messages/src/message.rs

```diff
@@ -114,7 +119,12 @@ pub enum ProtocolError {
 #[derive(Clone, Copy, Debug, Deserialize, PartialEq, Eq, Serialize, SerializedSize)]
 #[cfg_attr(any(test, feature = "std"), derive(thiserror::Error))]
 pub enum HwError {
-    /// The FPGA reported an I2C error
+    /// The FPGA reported an I2C error (deprecated)
+    ///
+    /// This error has been deprecated in favor of the more explicit and
+    /// informative errors below (NotPresent, NotPowered, PowerFault,
+    /// NotInitialized, I2cAddressNack, I2cByteNack). It is being maintained
+    /// for backwards compatability.
     #[cfg_attr(any(test, feature = "std"), error("FPGA reported an I2C error"))]
```

Comment:
I think we might want to [decorate this](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute) with `#[deprecated = "Use the more specific error variants instead"]` or similar.

Reply:
Good call!